modules/theAdxBidAdapter.js

```diff
@@ -0,0 +1,524 @@
+import * as utils from '../src/utils';
+import {
+  BANNER,
+  NATIVE,
+  VIDEO
+} from '../src/mediaTypes';
+import { config } from '../src/config';
+import * as url from '../src/url';
+import {
+  registerBidder
+} from '../src/adapters/bidderFactory';
+
+const BIDDER_CODE = 'theadx';
+const ENDPOINT_URL = '//ssp.theadx.com/request';
+const EVENT_PIXEL_URL = '//ssp.theadx.com/log';
+
+const NATIVEASSETNAMES = {
+  0: 'title',
+  1: 'cta',
+  2: 'icon',
+  3: 'image',
+  4: 'body',
+  5: 'sponsoredBy',
+  6: 'body2',
+  7: 'phone',
+  8: 'privacyLink',
+  9: 'displayurl',
+  10: 'rating',
+  11: 'address',
+  12: 'downloads',
+  13: 'likes',
+  14: 'price',
+  15: 'saleprice',
+
+};
+const NATIVEPROBS = {
+  title: {
+    id: 0,
+    name: 'title'
+  },
+  body: {
+    id: 4,
+    name: 'data',
+    type: 2
+  },
+  body2: {
+    id: 6,
+    name: 'data',
+    type: 10
+  },
+  privacyLink: {
+    id: 8,
+    name: 'data',
+    type: 501
+  },
+  sponsoredBy: {
+    id: 5,
+    name: 'data',
+    type: 1
+  },
+  image: {
+    id: 3,
+    type: 3,
+    name: 'img'
+  },
+  icon: {
+    id: 2,
+    type: 1,
+    name: 'img'
+  },
+  displayurl: {
+    id: 9,
+    name: 'data',
+    type: 11
+  },
+  cta: {
+    id: 1,
+    type: 12,
+    name: 'data'
+  },
+  rating: {
+    id: 7,
+    name: 'data',
+    type: 3
+  },
+  address: {
+    id: 11,
+    name: 'data',
+    type: 5
+  },
+  downloads: {
+    id: 12,
+    name: 'data',
+    type: 5
+  },
+  likes: {
+    id: 13,
+    name: 'data',
+    type: 4
+  },
+  phone: {
+    id: 7,
+    name: 'data',
+    type: 8
+  },
+  price: {
+    id: 14,
+    name: 'data',
+    type: 6
+  },
+  saleprice: {
+    id: 15,
+    name: 'data',
+    type: 7
+  },
+
+};
+
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: ['theadx'], // short code
+  supportedMediaTypes: [BANNER, VIDEO, NATIVE],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   *
+   * @param {BidRequest} bid The bid params to validate.
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    let res = false;
+    if (bid && bid.params) {
+      res = !!(bid.params.pid && bid.params.tagId && bid.params.url);
```

Comment:
You should require url to be a bidder parameter.  Please use bidderRequest.refererInfo instead
See http://prebid.org/dev-docs/bidder-adaptor.html#referrers

Reply:
url removed, we currently use refererInfo.